### DISCUSSION TITLE
关于"moduleResolution": "bundler",

### DISCUSSION THREAD

**Discussion by Cooper1886** [umijs/umi#11125] [Q&A]
请问下 我现在用pnpm创建的umi4项目，.umi>tsconfig.json中的配置是"moduleResolution": "bundler",我看了以前的项目是node，现在是bundler导致项目中import都报错，这个问题如何修改呢，手动修改的话，在删除.umi后重新start还是会设置成bundler。求教这个问题怎么解决呢？
![image](https://github.com/umijs/umi/assets/107901138/236ed56d-49ed-43b6-aa06-6658ca4b71e1)

**Comment by BoyYangzai** [comment]
@Johnnnnboy 
https://github.com/umijs/umi/blob/15d7385b88a2fd7fdbd8d93d23b35e5730a651f9/packages/preset-umi/src/features/tmpFiles/tmpFiles.ts#L61
看看是不是这个原因
这个tsconfig文件是根据你的项目配置自动生成的
![image](https://github.com/umijs/umi/assets/94534613/43658235-bd19-4eea-ae96-642cb2ce1790)
排查一下是不是这个原因

**Reply by Cooper1886** [reply]
项目中TS确实是5.0.3,如果是不想ts降级版本的话，就用bundler的话，怎么解决引用模块的问题呢？
而且现在创建umi4项目，package里面都是配置的ts5，这样的话每次都要手动改package吗？

**Reply by BoyYangzai** [reply]
@Johnnnnboy 
![image](https://github.com/umijs/umi/assets/94534613/16c8a8ee-5f83-4330-969f-d473b9256c80)
大佬的回答

**Reply by Cooper1886** [reply]
感谢

**Reply by jayconscious** [reply]
1
